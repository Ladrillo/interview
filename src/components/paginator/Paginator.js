import React, { Component } from 'react';
import styled from 'styled-components';


export default function Paginator({ upperLimit, onPrev, onNext, currentPage }) {
    const nextActive = currentPage < upperLimit;
    const prevActive = currentPage > 1;

    return (
        <Styled className='paginator' nextActive={nextActive} prevActive={prevActive}>
            <div onClick={onPrev} className='paginator-button prev'>Prev</div>
            <div onClick={onNext} className='paginator-button next'>Next</div>
        </Styled>
    );
}

const Styled = styled.div`
    width:           100%;
    display:         flex;
    justify-content: space-around;
    align-items:     center;
    margin-bottom:   20px;


    .paginator-button {
        border-radius:    4px;
        width:            80px;
        height:           30px;
        background-color: #222;
        display:          flex;
        justify-content:  center;
        align-items:      center;
        cursor:           pointer;

        &:hover {
            background-color: #494949;
        }
    }
`;