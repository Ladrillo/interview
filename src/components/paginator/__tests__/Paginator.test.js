import Paginator from '../Paginator';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';


describe('Paginator', () => {
    it('renders correctly', () => {
        const rendered = renderer.create(
            <Paginator
                onNext = {() => console.log('next!')}
                onPrev = {() => console.log('prev!')}
            />
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });
});
