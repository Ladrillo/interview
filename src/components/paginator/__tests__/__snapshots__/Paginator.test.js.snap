exports[`Paginator renders correctly 1`] = `
<div
  className="paginator sc-bdVaJa HJElk">
  <div
    className="paginator-button prev"
    onClick={[Function]}>
    Prev
  </div>
  <div
    className="paginator-button next"
    onClick={[Function]}>
    Next
  </div>
</div>
`;
